scripts/upload.js

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const childProcess = require('child_process')
+const commander = require('commander')
+const crypto = require('crypto')
+const fs = require('fs')
+const mkdirp = require('mkdirp')
+const path = require('path')
+
+process.on('uncaughtException', (err) => {
+  console.error('Caught exception:', err)
+  process.exit(1)
+})
+
+process.on('unhandledRejection', (err) => {
+  console.error('Unhandled rejection:', err)
+  process.exit(1)
+})
+
+function createManifestForUpload (commander) {
+  const componentManifest = path.join('manifests', commander.type + '-manifest.json')
+  const name = JSON.parse(fs.readFileSync(componentManifest)).name
+
+  const manifestDir = path.join('build', commander.type, 'stable', 'extensions')
+  const manifestFile = path.join(manifestDir, 'extensionManifest.json')
+
+  const content =
+        [
+          [
+            getIDFromComponentType(commander.type),
+            commander.setVersion,
+            generateSHA256HashOfFile(commander.crx),
+            name
+          ]
+        ]
+
+  mkdirp.sync(manifestDir)
+
+  fs.writeFileSync(manifestFile, JSON.stringify(content, null, 2))
+}
+
+function generateSHA256HashOfFile (file) {
+  const data = fs.readFileSync(file)
+  const hash = crypto.createHash('sha256')
+  return hash.update(data).digest('hex')
+}
+
+function getIDFromComponentType (componentType) {
```

Comment:
let's try to get rid of that and get the extension ID from the private key instead with a method similar to:`script/get-extension-public-key-sha256.py`

Reply:
Great idea! Fixed in latest commit, please let me know if there's a more idiomatic way to do the hex translation.